src/verifiers/sev-snp/crypto.c

```diff
@@ -0,0 +1,175 @@
+/* Copyright (c) 2021 Intel Corporation
+ * Copyright (c) 2020-2021 Alibaba Cloud
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <openssl/sha.h>
+#include <openssl/ts.h>
+#include <rats-tls/log.h>
+#include <string.h>
+#include "crypto.h"
+
+static bool digest_sha(const void *msg, size_t msg_len, uint8_t *digest, size_t digest_len,
+		       SHA_TYPE sha_type)
+{
+	if (sha_type == SHA_TYPE_256 && digest_len != SHA256_DIGEST_LENGTH)
+		return false;
+
+	if (sha_type == SHA_TYPE_256) {
+		SHA256_CTX context;
+
+		if (SHA256_Init(&context) != 1)
+			return false;
+		if (SHA256_Update(&context, (void *)msg, msg_len) != 1)
+			return false;
+		if (SHA256_Final(digest, &context) != 1)
+			return false;
+	} else if (sha_type == SHA_TYPE_384) {
+		SHA512_CTX context;
+
+		if (SHA384_Init(&context) != 1)
+			return false;
+		if (SHA384_Update(&context, (void *)msg, msg_len) != 1)
+			return false;
+		if (SHA384_Final(digest, &context) != 1)
+			return false;
+	}
+
+	return true;
+}
+
+static bool ecdsa_verify(sev_sig *sig, EVP_PKEY **pub_evp_key, uint8_t *digest, size_t length)
```

Comment:
ditto.

These functions should be putted in openssl crypto instance and then call them through crypto interface.